linux/ibus-keyman/src/engine.c

```diff
@@ -831,12 +915,14 @@ ibus_keyman_engine_process_key_event(
 
   // km_kbp_state_action_items to get action items
   size_t num_action_items;
-  g_free(keyman->char_buffer);
-  keyman->char_buffer                    = NULL;
+  g_free(keyman->commit_item->char_buffer);
+  keyman->commit_item->char_buffer                    = NULL;
   const km_kbp_action_item *action_items = km_kbp_state_action_items(keyman->state, &num_action_items);
 
-  if (!process_actions(engine, action_items, num_action_items))
+  if (!process_actions(engine, action_items, num_action_items) &&
+      (!(engine->client_capabilities & IBUS_CAP_PREFILTER) || (engine->client_capabilities & IBUS_CAP_SURROUNDING_TEXT))) {
     return FALSE;
```

Comment:
We should document why we return FALSE only when the !PREFILTER/SURROUNDING_TEXT capability condition is met.

Reply:
done